src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2514,22 +2514,42 @@ internal override Symbol RemapSymbolIfNecessaryCore(Symbol symbol)
 
         internal override bool ShouldSkipSyntaxNodeAnalysis(SyntaxNode node, ISymbol containingSymbol)
         {
-            if (containingSymbol.Kind is SymbolKind.Method)
+            switch (containingSymbol.Kind)
             {
-                switch (node)
-                {
-                    case TypeDeclarationSyntax:
-                        // Skip the topmost type declaration syntax node when analyzing primary constructor
-                        // to avoid duplicate syntax node callbacks.
-                        // We will analyze this node when analyzing the type declaration type symbol.
-                        return true;
+                case SymbolKind.Method:
+                    switch (node)
+                    {
+                        case TypeDeclarationSyntax:
+                            // Skip the topmost type declaration syntax node when analyzing primary constructor
+                            // to avoid duplicate syntax node callbacks.
+                            // We will analyze this node when analyzing the type declaration type symbol.
+                            return true;
+
+                        case CompilationUnitSyntax:
+                            // Skip compilation unit syntax node when analyzing synthesized top level entry point method
+                            // to avoid duplicate syntax node callbacks.
+                            // We will analyze this node when analyzing the global namespace symbol.
+                            return true;
+
+                        case BaseListSyntax:
```

Comment:
>BaseListSyntax [](http://example.com/codeflow?start=29&length=14)

Is it possible to subscribe to an event for items other that the first one in this list? How many call backs are we going to get for them in presence of a primary constructor?

Reply:
What about other parts of the type declaration header?

Reply:
Enhanced the added unit test